### PR TITLE
[#113] : fix Utils#getField to not log error when field is on a superclass

### PR DESCRIPTION
**Pull request by massifben**
Signed-off-by: massifben <105049157+massifben@users.noreply.github.com>

### CODE REVIEW

File: sct-commons/src/test/java/org/lfenergy/compas/sct/commons/UtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        // Given : nothing
+        // When
+        Field childField = Utils.getField(ChildClass.class, "childField");
+        // Then
+        assertNotNull(childField);
+        assertEquals("childField", childField.getName());
+    }
+
+    @Test
+    void getField_should_return_declaredField_when_declared_on_superclass() {
```
Comment (psabatier74) [inline sct-commons/src/test/java/org/lfenergy/compas/sct/commons/UtilsTest.java:26]: Naming suggestion : should_look_in_parent_classes_to_find_field()

File: sct-commons/src/test/java/org/lfenergy/compas/sct/commons/UtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Field;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class UtilsTest {
+
+    @Test
+    void getField_should_return_declaredField() {
```
Comment (psabatier74) [inline sct-commons/src/test/java/org/lfenergy/compas/sct/commons/UtilsTest.java:16]: Naming suggestion : should_find_field_in_current_class()

File: sct-commons/src/test/java/org/lfenergy/compas/sct/commons/UtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    void setField_should_do_nothing_when_field_not_found() {
+        // Given
+        ChildClass child = new ChildClass();
+        // When & Then
+        assertDoesNotThrow(() ->
+            Utils.setField(child, "non existent field", "changed"));
+    }
+
+    @Test
+    void setField_should_do_nothing_when_value_is_not_assignable() {
```
Comment (psabatier74) [inline sct-commons/src/test/java/org/lfenergy/compas/sct/commons/UtilsTest.java:84]: why value could not be assignable ?

Comment (AliouDIAITE) [reply sct-commons/src/test/java/org/lfenergy/compas/sct/commons/UtilsTest.java:84]: Suggest as there is question about test : 
setField_should_do_nothing_when_value_types_mismatch

**Comment by SaintierFr** [comment]
In order to run UtilsTest locally, I have to upgrade the commons-lang3 version to 3.12.0 in sct-commons POM, don't you? (the "mvn test" don't need this upgrade to work)

File: sct-commons/src/test/java/org/lfenergy/compas/sct/commons/UtilsTest.java (hunk 3 above)
Comment (massifben) [reply sct-commons/src/test/java/org/lfenergy/compas/sct/commons/UtilsTest.java:84]: OK. I use the suggestion of @AliouDIAITE.

**Pull request closed without merging**